packages/client/src/internals/dataLoader.ts

```diff
@@ -69,7 +72,15 @@ export function dataLoader<TKey, TValue>(fetchMany: BatchLoadFn<TKey, TValue>) {
       item.reject = reject;
       item.resolve = resolve;
       thisBatch.items.push(item);
+
+      if (
+        typeof opts?.maxBatchSize !== 'undefined' &&
+        thisBatch.items.length >= opts.maxBatchSize
+      ) {
+        // TODO: dispatch... but how?
+      }
     });
+
```

Comment:
Dispatch after the promise

```suggestion
    });
    if (
      typeof opts?.maxBatchSize !== 'undefined' &&
      thisBatch.items.length >= opts.maxBatchSize
    ) {
      dispatch();
      return;
    }
```